### PR TITLE
optimize: change "IsPersist" property value type from String to Boolean

### PR DESCRIPTION
**Pull request by long187**
### Ⅰ. Describe what this PR did
Saga: Change "IsPersist" property value type from String to Boolean

### Ⅱ. Does this pull request fix one issue?
#1827

### CODE REVIEW

File: saga/seata-saga-statelang/src/main/java/io/seata/saga/statelang/parser/impl/StateMachineParserImpl.java [hunk 1]
```diff
@@ -43,7 +43,8 @@ public StateMachine parse(String json) {
         stateMachine.setComment((String)node.get("Comment"));
         stateMachine.setVersion((String)node.get("Version"));
         stateMachine.setStartState((String)node.get("StartState"));
-        if("false".equals(node.get("IsPersist"))){
+        Object isPersist = node.get("IsPersist");
+        if(isPersist != null && Boolean.FALSE.equals(isPersist)){
```
Comment (xingfudeshi) [inline saga/seata-saga-statelang/src/main/java/io/seata/saga/statelang/parser/impl/StateMachineParserImpl.java]: if(Boolean.FALSE.equals(isPersist))

File: saga/seata-saga-statelang/src/main/java/io/seata/saga/statelang/parser/impl/AbstractTaskStateParser.java [hunk 2]
```diff
@@ -41,7 +41,8 @@ protected void parseTaskAttributes(AbstractTaskState state, Object node) {
         state.setCompensateState((String) nodeMap.get("CompensateState"));
         state.setForCompensation("true".equals(nodeMap.get("IsForCompensation")));
         state.setForUpdate("true".equals(nodeMap.get("IsForUpdate")));
-        if("false".equals(nodeMap.get("IsPersist"))){
+        Object isPersist = nodeMap.get("IsPersist");
+        if(isPersist != null && Boolean.FALSE.equals(isPersist)){
```
Comment (xingfudeshi) [inline saga/seata-saga-statelang/src/main/java/io/seata/saga/statelang/parser/impl/AbstractTaskStateParser.java]: if(Boolean.FALSE.equals(isPersist))

File: saga/seata-saga-statelang/src/main/java/io/seata/saga/statelang/parser/impl/StateMachineParserImpl.java (hunk 1 above)
Comment (long187) [reply saga/seata-saga-statelang/src/main/java/io/seata/saga/statelang/parser/impl/StateMachineParserImpl.java]: fixed

File: saga/seata-saga-statelang/src/main/java/io/seata/saga/statelang/parser/impl/AbstractTaskStateParser.java (hunk 2 above)
Comment (long187) [reply saga/seata-saga-statelang/src/main/java/io/seata/saga/statelang/parser/impl/AbstractTaskStateParser.java]: fixed

**Pull request merged**